gpu/gpu_darwin.go

```diff
@@ -15,19 +19,9 @@ func CheckVRAM() (int64, error) {
 		// gpu not supported, this may not be metal
 		return 0, nil
 	}
-
-	// on macOS, there's already buffer for available vram (see below) so just return the total
-	systemMemory := int64(memory.TotalMemory())
-
-	// macOS limits how much memory is available to the GPU based on the amount of system memory
-	// TODO: handle case where iogpu.wired_limit_mb is set to a higher value
-	if systemMemory <= 36*1024*1024*1024 {
-		systemMemory = systemMemory * 2 / 3
-	} else {
-		systemMemory = systemMemory * 3 / 4
-	}
-
-	return systemMemory, nil
+	recommendedMaxVRAM := int64(C.getRecommendedMaxVRAM())
+	slog.Info(fmt.Sprintf("Recommended max VRAM: %d MB", recommendedMaxVRAM/1024/1024))
```

Comment:
Thanks! I think we'd want to log it here https://github.com/ollama/ollama/blob/main/llm/llm.go#L48 and let the caller of this function "decide" whether it should be logged

Reply:
Okay, then I remove logging from gpu_darwin.go.

I also checked twice whether I got the memory management correctly. Turns out that Core Foundation objects still need manual releasing in Objective-C, so I fixed that.